nilearn/load_confounds/parser.py

```diff
@@ -0,0 +1,387 @@
+"""Flexible method to load confounds generated by fMRIprep.
+
+Authors: load_confounds team
+"""
+import numpy as np
+import pandas as pd
+from . import confounds as cf
+from .compcor import _find_compcor
+
+# Global variables listing the admissible types of noise components
+all_confounds = [
+    "motion",
+    "high_pass",
+    "wm_csf",
+    "global",
+    "compcor",
+    "ica_aroma",
+    "scrub",
+    "non_steady_state",
+]
+
+
+def _sanitize_strategy(strategy):
+    """Defines the supported denoising strategies."""
+    if not isinstance(strategy, list):
+        raise ValueError("strategy needs to be a list of strings")
+    for conf in strategy:
+        if conf not in all_confounds:
+            raise ValueError(f"{conf} is not a supported type of confounds.")
+    # add non steady state if not present
+    if "non_steady_state" not in strategy:
+        strategy.append("non_steady_state")
+    return strategy
+
+
+def _check_error(missing_confounds, missing_keys):
+    """Consolidate a single error message across multiple missing confounds."""
+    if missing_confounds or missing_keys:
+        error_msg = (
+            "The following keys or parameters are missing: "
+            + f" {missing_confounds}"
+            + f" {missing_keys}"
+            + ". You may want to try a different denoising strategy."
+        )
+        raise ValueError(error_msg)
+
+
+class Confounds:
```

Comment:
I'm wondering why it is a class not a function ?
Does it have some internal variables / state that justify having a class ?
To me it follows an input/output logic, i.e. that of a function, but I may have missed something.

Reply:
Here is a piece of important code that's not in this PR:
we have some predefined strategy that inherent the Confounds class: https://github.com/SIMEXP/load_confounds/blob/master/load_confounds/strategies.py
The main motivation is to cut down the parameters exposed and some internal sanity check between different noise components. If we write this as a function, each helper function will have to take more parameters than now. 
Happy to discuss!

Reply:
I don't fully undestand. You could perfectly have strategies in a function ?